backend/app/api/managers.py

```diff
@@ -0,0 +1,32 @@
+from typing import List
+
+from fastapi import WebSocket
+
+from app.config import settings
+
+
+class WsConnectionManager:
+    def __init__(self) -> None:
+        self.active_connections: dict[int, List[WebSocket]] = {}
+
+    async def connect(self, websocket: WebSocket, room_id: int) -> None:
+        if (
+            self.active_connections.get(room_id)
+            and len(self.active_connections[room_id]) == settings.max_ws_connections
```

Comment:
What happens if, say, client reloads their page? As far as I understand, the websocket connection should break. Does the websocket object get deleted from the corresponding list (so that a new connection can be established)? What part of code does that?

Reply:
When a websocket connection is broken, `WebSocketDisconnect` exception is raised, and this exception is handled in the websocket router and the corresponding websocket object is deleted by `disconnect` method in `WsConnectionManager`